src/NServiceBus.Transport.Msmq/Persistence/SubscriptionStorage/MsmqSubscriptionPersistence.cs

```diff
@@ -9,15 +11,24 @@ class MsmqSubscriptionPersistence : Feature
     {
         protected override void Setup(FeatureConfigurationContext context)
         {
-            var queueName = context.Settings.GetConfiguredMsmqPersistenceSubscriptionQueue();
+            var configuredQueueName = context.Settings.GetConfiguredMsmqPersistenceSubscriptionQueue();
 
-            if (string.IsNullOrEmpty(queueName))
+            if (string.IsNullOrEmpty(configuredQueueName))
             {
-                queueName = "NServiceBus.Subscriptions";
-                Logger.Warn($"The queue used to store subscriptions has not been configured, so the default '{queueName}' will be used.");
+                if (DoesOldDefaultQueueExists())
+                {
+                    // The user has not configured the subscriptions queue to be "NServiceBus.Subscriptions" but there's a local queue. 
+                    // Indicates that the endoint was using old default queue name.
+                    throw new Exception(
+                        "Detected the presence of an old default queue named `NServiceBus.Subscriptions`. The new default is now `[Your endpointname].Subscriptions`. Move the messages to the new queue or configure the subscriptions queue name.");
+                }
+
+                var defaultQueueName = $"{context.Settings.EndpointName()}.Subscriptions";
```

Comment:
Call me old fashioned, but I would prefer the following, the current LOC makes me think.

    var defaultQueueName = context.Settings.EndpointName() + ".Subscriptions";

Reply:
@ramonsmits - Yep. You're old fashioned :)